cf/mixin/fielddomain.py

```diff
@@ -0,0 +1,2958 @@
+        differences) and ``rtol`` (the tolerance on relative differences)
+        are positive, typically very small numbers. See the *atol* and
+        *rtol* parameters.
+
+        :Parameters:
+
+            ref0: `CoordinateReference`
+
+            ref1: `CoordinateReference`
+
+            field1: `Field`
+                The field which contains *ref1*.
+
+        :Returns:
+
+            `bool`
+
+        """
+        ref0 = self.coordinate_references(todict=True)[key0]
+        ref1 = field1.coordinate_references(todict=True)[key1]
+
+        if not ref0.equivalent(ref1, rtol=rtol, atol=atol, verbose=verbose):
+            logger.info(
+                f"{self.__class__.__name__}: Non-equivalent coordinate "
+                f"references ({ref0!r}, {ref1!r})"
+            )  # pragma: no cover
+            return False
+
+        # Compare the domain ancillaries
+        # TODO consider case of None key ?
+        for (
+            term,
+            identifier0,
+        ) in ref0.coordinate_conversion.domain_ancillaries().items():
+            if identifier0 is None:
+                continue
+
+            identifier1 = ref1.coordinate_conversion.domain_ancillaries()[term]
+
+            #            key0 = domain_ancillaries.filter_by_key(identifier0).key()
+            #            key1 = field1_domain_ancillaries.filter_by_key(identifier1).key()
+
+            if not self._equivalent_construct_data(
+                field1,
+                key0=identifier0,  # key0,
+                key1=identifier1,  # key1,
+                rtol=rtol,
+                atol=atol,
+                s=s,
+                t=t,
+                verbose=verbose,
+                axis_map=axis_map,
+            ):
+                # add traceback TODO
+                return False
+
+        return True
+
+    def _indices(self, mode, data_axes, auxiliary_mask, **kwargs):
+        """Create indices that define a subspace of the field or domain
+        construct.
+
+        This method is intended to be called by the `indices` method.
+
+        See the `indices` method for more details.
+
+        .. versionadded:: 3.9.0
+
+        :Parameters:
+
+            mode: `str`
+                The mode of operation. See the *mode* parameter of
+                `indices` for details.
+
+            data_axes: sequence of `str`, or `None`
+                The domain axis identifiers of the data axes, or
+                `None` if there is no data array.
+
+            auxiliary_mask: `bool`
+                Whether or not to create an auxiliary mask. See
+                `indices` for details.
+
+            kwargs: *optional*
+                See the **kwargs** parameters of `indices` for
+                details.
+
+        :Returns:
+
+            `dict`
+                 The dictionary has two keys: ``'indices'`` and
+                 ``'mask'``.
+
+                 The ``'indices'`` key stores a dictionary in keyed by
```

Comment:
What is meant by "in keyed" (also used in the next paragraph)?

Reply:
Sorry - in the "in" is rogue. Should be 
```
                 The ``'indices'`` key stores a dictionary keyed by
                 domain axis identifiers, each of which has a value of
                 the index for that domain axis.
```